### ISSUE TITLE
enable-auto-commit is not working

### ISSUE THREAD

**Issue by rotilho** [spring-cloud/spring-cloud-stream-binder-kafka#124]
The new Spring Boot property spring.kafka.consumer.enable-auto-commit is ignored.

Expected behavour:
When spring.kafka.consumer.enable-auto-commit=false the Message Header should contains Acknowledgment

Workaround:
The autoCommitOffset=false make it work as expected

**Comment by rotilho** [comment]
@ilayaperumalg do you think this is a bug?

**Comment by garyrussell** [comment]
These are two different properties.

The first is a kafka client property - the kafka client takes care of the commits itself.

The binder uses the spring-kafka `ConcurrentMessageListenerContainer` - it has acknowledge modes where the container can commit the offset after the listener has processed a record. You could call it "container commit mode".

Only when using a `MANUAL` commit mode does the `Acknowlegment` appear as a header - in MANUAL modes, the user is responsible for acking the message and the container won't do it for you.

In many cases, the container commit modes are all that's needed.

The binder `autoCommitOffset` property (when `false`) sets the containers commit (ack) mode to `MANUAL`.

**Comment by rotilho** [comment]
@garyrussell thank for the explanation but still think it's the same.

Let me know if I'm wrong: Setting the commit to MANUAL in Spring Kafka (enable-auto-commit=false) or Spring Cloud Streams Kafka (autoCommitOffset=false) allows the consumer to Acknowledge the message.

Knowing this, shouldn't Spring Cloud Streams take in consideration enable-auto-commit from Spring Boot when creating the Consumer?

**Comment by garyrussell** [comment]
No; as I said they are different.

`enable-auto-commit` maps to the kafka consumer property `enable.auto.commit`. When true, the client itself commits offsets (on a schedule).

The container has a bunch of ack modes which apply when `enable.auto.commit` is `false` (RECORD, BATCH, ..., MANUAL).

The binder currently does not expose all those modes, just a simple boolean `autoCommitOffset` which sets the AckMode to MANUAL (instead of the default, BATCH).

In summary, there are 3 types of offset commits

- client acks on its own
- container acks on behalf of the listener (using several different options)
- application acks - enabled by MANUAL ackmode on the container.

>Knowing this, shouldn't Spring Cloud Streams take in consideration enable-auto-commit from Spring Boot when creating the Consumer?

Since it means something different, we can't assume the container mode from the client boolean.

**Comment by rotilho** [comment]
Clear. Thank you again!
